### PR TITLE
backport: merge bitcoin#20332, #20908, #21336, #21443, #21142, #21512, #20833, #21522, #21553, #21970, #24336, partial bitcoin#21606 (fuzzing harness backports: part 7)

### PR DESCRIPTION
**Pull request by kwvg**
## Additional Information

* Depends on https://github.com/dashpay/dash/pull/5834

### CODE REVIEW

**Review by UdjinM6 — APPROVED** [review]
utACK

File: src/test/fuzz/tx_pool.cpp [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+    const auto& node = g_setup->m_node;
+    auto& chainstate = node.chainman->ActiveChainstate();
+
+    SetMockTime(ConsumeTime(fuzzed_data_provider));
+    SetMempoolConstraints(*node.args, fuzzed_data_provider);
+
+    // All RBF-spendable outpoints
+    std::set<COutPoint> outpoints_rbf;
```
[the remark this answers is not in the document]
Comment (kwvg) [reply src/test/fuzz/tx_pool.cpp:89]: The RBF-friendly logic doesn't seem to really be a problem, I don't remember having to remove any RBF-specific logic from it. It does fuzz some `CTxMempool` and transaction package logic and for that alone, retaining the tests seem valuable enough.

File: src/test/fuzz/coins_view.cpp [hunk 2]
```diff
@@ -229,11 +229,8 @@ FUZZ_TARGET_INIT(coins_view, initialize_coins_view)
                     // consensus/tx_verify.cpp:171: bool Consensus::CheckTxInputs(const CTransaction &, TxValidationState&, const CCoinsViewCache &, int, CAmount &): Assertion `!coin.IsSpent()' failed.
                     return;
                 }
-                try {
-                    (void)Consensus::CheckTxInputs(transaction, state, coins_view_cache, fuzzed_data_provider.ConsumeIntegralInRange<int>(0, std::numeric_limits<int>::max()), tx_fee_out);
-                    assert(MoneyRange(tx_fee_out));
-                } catch (const std::runtime_error&) {
-                }
+                (void)Consensus::CheckTxInputs(transaction, state, coins_view_cache, fuzzed_data_provider.ConsumeIntegralInRange<int>(0, std::numeric_limits<int>::max()), tx_fee_out);
```
[the remark this answers is not in the document]
Comment (kwvg) [reply src/test/fuzz/coins_view.cpp:238]: Good catch, resolved in latest push!

**Review by knst — APPROVED** [review]
utACK

**Review by UdjinM6 — APPROVED** [review]
re-utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**